opentelemetry/proto/profiles/v1development/profiles.proto

```diff
@@ -251,9 +251,11 @@ message Profile {
   int32 default_sample_type_index = 9;
 
   // A globally unique identifier for a profile. The ID is a 16-byte array. An ID with
-  // all zeroes is considered invalid.
-  //
-  // This field is required.
+  // all zeroes is considered invalid. It may be used for deduplication and signal
+  // correlation purposes. It is acceptable to treat two profiles with different values
+  // in this field as not equal, even if they represented the same object at an earlier
+  // time.
+  // This field is optional; an ID may be assigned to an ID-less profile in a later step.
```

Comment:
What value is it set to if unset and optional?

Reply:
Right, that's a bit unclear, let me rephrase.

> The value would be a real valid identifier, including 16 zeros if that makes sense.

The receiver of an id-less profile (i.e. `profile_id` absent) is allowed to assign an arbitrary value to `profile_id`, including the null value (16 zeros).